core/shared/src/main/scala/io.buildo.metarpheus/core/Metarpheus.scala

```diff
@@ -2,12 +2,26 @@ package io.buildo.metarpheus
 package core
 
 import scala.meta._
-import scala.meta.inputs.Input
+import scala.meta.inputs.Input.File
+import scala.meta.io.AbsolutePath
+import scala.meta.internal.io.PlatformFileIO
 
 object Metarpheus {
 
-  def run(files: List[String], config: Config): intermediate.API = {
-    val parsed = files.map(Input.String(_).parse[Source].get)
+  def recursivelyListFiles(target: AbsolutePath): List[AbsolutePath] = {
+    if (target.isDirectory) {
+      val dirContent = PlatformFileIO.listFiles(target).iterator.toList
+      dirContent ++ dirContent.filter(_.isDirectory).flatMap(recursivelyListFiles)
```

Comment:
`.filter(_.isDirectory)` seems just an (unnecessary?) optimization.

Moreover `val dirContent = PlatformFileIO.listFiles(target).iterator.toList` is not filtering out directories, so you end up with a List containing bot directories and, recursively, the files contained in them: is this the expected behaviour?

For example it's not clear what happens if the initial input is an empty directory: you have `Nil` or a `List(directory)`?